app/views/advanced_messenger/_history_journal_bottom.html.erb

```diff
@@ -0,0 +1,151 @@
+<%= javascript_tag do %>
+// We need to enclose the code in a block {} because otherwise we get a javascript compilation error about variables/constants being declared many times
+{
+    const currentUserId = "<%= User.current.id %>";
+    const RED = "red";
+    const GREEN = "#55CB85";
+
+    function renderCollapsed(id, isOfInterestForCurrentUser, isInitialRender) {
+        var currentNote = $("#change-" + id + ".journal.has-notes");
```

Comment:
Vad ca bucatica asta de cod pare sa fie cam la fel (cu exceptia `icon-toggle-plus` vs `icon-toggle-minus`). Ma intreb daca nu s-ar putea externaliza intr-o functie

Reply:
Ok. I extracted the duplicated code to a function